cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/consumption/AwsEBSConsumptionCalculator.java

```diff
@@ -0,0 +1,115 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.consumption;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.validation.ValidationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.ec2.model.DescribeVolumesResult;
+import com.amazonaws.services.ec2.model.Volume;
+import com.cloudera.thunderhead.service.metering.events.MeteringEventsProto;
+import com.sequenceiq.cloudbreak.cloud.ConsumptionCalculator;
+import com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource.AwsEbsCommonService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudConsumption;
+import com.sequenceiq.cloudbreak.cloud.model.StorageSizeRequest;
+import com.sequenceiq.cloudbreak.cloud.model.StorageSizeResponse;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.common.mappable.StorageType;
+
+@Service
+public class AwsEBSConsumptionCalculator implements ConsumptionCalculator {
+
+    private static final StorageType EBS = StorageType.EBS;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsEBSConsumptionCalculator.class);
+
+    private static final long NO_BYTE_IN_MB = 1000L * 1000L;
+
+    private static final int DATE_RANGE_WIDTH_IN_DAYS = 2;
+
+    @Inject
+    private AwsEbsCommonService awsEbsCommonService;
+
+    @Override
+    public void validate(CloudConsumption cloudConsumption) throws ValidationException {
+        String storageLocation = cloudConsumption.getStorageLocation();
+        if (storageLocation == null || !storageLocation.startsWith("vol-")) {
+            throw new ValidationException(String.format("EBS id must start with 'vol-' if required file system type is 'EBS'!"));
+        }
+    }
+
+    @Override
+    public StorageSizeResponse calculate(StorageSizeRequest request) {
+        DescribeVolumesResult result = awsEbsCommonService.getEbsSize(
```

Comment:
I think we discussed that in case of datahub crn we will go to CB service for get Datahub informations